File: WandleWheelhouse/src/WandleWheelhouse.Api/DTOs/Subscriptions/SubscriptionResponseDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using System;
using WandleWheelhouse.Api.Models;

namespace WandleWheelhouse.Api.DTOs.Subscriptions;

#nullable enable
public class SubscriptionResponseDto
{
    public Guid SubscriptionId { get; set; }
    public decimal MonthlyAmount { get; set; }
    public PaymentMethod Method { get; set; }
    public SubscriptionStatus Status { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime? NextPaymentDate { get; set; }
    public DateTime? CancellationDate { get; set; }
    public string? ProviderSubscriptionId { get; set; } // ID from PayPal/Worldpay

    // User Info (Consider if needed, adds complexity if User not loaded)
    public string UserId { get; set; } = string.Empty;
    public string? UserFullName { get; set; }
}
#nullable disable

File: WandleWheelhouse/client/src/dto/Subscriptions/SubscriptionResponseDto.ts
import { PaymentMethod } from '../Donations/PaymentMethodEnum'; // Reuse enum
import { SubscriptionStatus } from './SubscriptionStatusEnum';

// Matches backend DTO for displaying subscription details
export interface SubscriptionResponseDto {
  subscriptionId: string; // GUID as string
  monthlyAmount: number;
  method: PaymentMethod;
  status: SubscriptionStatus;
  startDate: string; // ISO Date string
  nextPaymentDate?: string | null; // ISO Date string or null
  cancellationDate?: string | null; // ISO Date string or null (Now indicates when it WILL be cancelled if pending)
  providerSubscriptionId?: string | null;

  // User Info (Included from backend if needed, e.g., for Admin view)
  userId: string;
  userFullName?: string | null;
}